File: typescript/src/llm/response.ts
export interface ToolCall {
  id: string;
  name: string;
  arguments: Record<string, unknown>;
}

export interface TokenUsage {
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
}

export interface Response {
  content: string;
  /** Raw reasoning returned on a separate channel (e.g. reasoning_content for
   * DeepSeek-style models). Never written into content, never returned to the
   * end user, and never recorded on a default audit record; consumers
   * normalize it with the reasoning adapter. */
  reasoning: string;
  tool_calls: ToolCall[];
  finish_reason: string;
  usage_tokens: TokenUsage;
}

File: go/llm/response.go
package llm

// Response is the output from a completion.
type Response struct {
	Content string
	// Reasoning is the raw reasoning the backend returned on a separate
	// channel (e.g. reasoning_content for DeepSeek-style models) when the
	// backend captures it. It is never written into Content, never returned to
	// the end user, and never recorded on a default audit record; consumers
	// normalize it with the reasoning adapter (go/reasoning).
	Reasoning    string
	ToolCalls    []ToolCall
	FinishReason string
	UsageTokens  TokenUsage
}

// ToolCall is a structured tool invocation from the LLM.
type ToolCall struct {
	ID   string
	Name string
	Args map[string]any
}

// TokenUsage tracks prompt and completion token counts.
type TokenUsage struct {
	PromptTokens     int
	CompletionTokens int
	TotalTokens      int
	// CachedTokens is the subset of PromptTokens served from the provider's
	// prompt cache. Zero when the backend does not report cache usage.
	CachedTokens int
}
